cmake/compilers/Clang.cmake

```diff
@@ -12,6 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if(CMAKE_CXX_COMPILER MATCHES "/em\\+\\+(-[a-zA-Z0-9.])?$")
+  message(" * C++ compiler: Emscripten")
+  set(CMAKE_EXECUTABLE_SUFFIX .js)
```

Comment:
Another reason to use `emcmake` to build: it sets the executable suffix to `.js` for you.